presto-tests/src/main/java/io/prestosql/tests/TestingPrestoClient.java

```diff
@@ -238,10 +238,12 @@ else if (type instanceof ArrayType) {
                     .collect(toList());
         }
         else if (type instanceof MapType) {
-            return ((Map<Object, Object>) value).entrySet().stream()
-                    .collect(Collectors.toMap(
-                            e -> convertToRowValue(((MapType) type).getKeyType(), e.getKey()),
-                            e -> convertToRowValue(((MapType) type).getValueType(), e.getValue())));
+            Map<Object, Object> result = new HashMap<>();
+            ((Map<Object, Object>) value)
+                    .forEach((k, v) -> result.put(
```

Comment:
It's more idiomatic to do it in pure functional style than via side-effects:

```java
return ((Map<Object, Object>) value)
        .entrySet().stream()
        .collect(Collectors.toMap(
                k -> convertToRowValue(((MapType) type).getKeyType(), k),
                v -> convertToRowValue(((MapType) type).getValueType(), v)));
```

Reply:
It was written in functional style previously, but the collector didn't accept `null` values. I needed them to test `hstore` who permits `null` values.

Reply:
Good point. Interestingly, `Collectors.toMap` allows null keys but not null values.